src/Utilities/PointerVector.hpp

```diff
@@ -23,23 +23,56 @@
 #define SPECTRE_BLAZE_DEALLOCATOR blaze::ArrayDelete()
 
 namespace blaze {
+
 template <typename T>
 BLAZE_ALWAYS_INLINE SIMDdouble step_function(const SIMDf64<T>& v) noexcept
 #if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
 {
-  return _mm512_set_pd((~v)[7] < 0.0 ? 0.0 : 1.0, (~v)[6] < 0.0 ? 0.0 : 1.0,
-                       (~v)[5] < 0.0 ? 0.0 : 1.0, (~v)[4] < 0.0 ? 0.0 : 1.0,
-                       (~v)[3] < 0.0 ? 0.0 : 1.0, (~v)[2] < 0.0 ? 0.0 : 1.0,
-                       (~v)[1] < 0.0 ? 0.0 : 1.0, (~v)[0] < 0.0 ? 0.0 : 1.0);
+  // A step function entirely in terms of intel vectorization instructions
+  // first, we `reinterpret_cast` to an integer type, as the bit operations
+  // are easier as unsigned ints.
+  //
+  // Then, we take the `_mmxxx_andnot_pd` = (~a) & 0x8000 0000 0000 0000, which
+  // extracts the sign bit and inverts it. So, if we started with a negative
+  // value (1 in the first bit), the new value will have 0's in all 64
+  // bits. Otherwise, it is 1 followed by 64 zeros
+  // 1000 0000 0000 0000  0000 0000 0000 0000 ...(32 more zeros)
+  //
+  // all remaining operations preserve the all-zeros in the first case, so
+  // concentrate on the second case. We right-shift by 32, giving
+  // 0000 0000 0000 0000  0000 0000 0000 0000  1000 ...(24 more zeros)
+  // finally, we multiply least-significant 32-bit blocks and store them
```

Comment:
finally -> Finally

I found this sentence confusing. Are you multiplying "(least significant 32 bits) * (0111 1111.... etc)" ? If so I would reorder clauses to be "we multiply the least-significant 32-bit blocks with the constant 32-bit input ___ and store the result in 64-bit blocks"... or similar. I'm trying to provide a helpful suggestion but wasn't actually sure of what you were doing here :P